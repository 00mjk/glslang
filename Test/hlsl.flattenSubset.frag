struct S0
{
    int x;
    int y;
    SamplerState ss;
};

struct S1
{
    float b;
    SamplerState samplerState;
    S0 s0;
    int a;
};

struct S2
{
    int a1;
    int a2;
    int a3;
    int a4;
    int a5;
    S1 resources;
};

float4 main(float4 vpos : VPOS) : COLOR0
{
    S1 s1;
    S2 s2;
    s2.resources = s1;
    return float4(0,0,0,0);
}
