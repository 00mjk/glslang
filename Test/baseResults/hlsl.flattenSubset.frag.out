hlsl.flattenSubset.frag
WARNING: AST will form illegal SPIR-V; need to transform to legalize
Shader version: 500
gl_FragCoord origin is upper left
0:? Sequence
0:27  Function Definition: @main(vf4; ( temp 4-component vector of float)
0:27    Function Parameters: 
0:27      'vpos' ( in 4-component vector of float)
0:?     Sequence
0:30      Sequence
0:30        move second child to first child ( temp float)
0:?           's2.resources.b' ( temp float)
0:?           's1.b' ( temp float)
0:30        move second child to first child ( temp sampler)
0:?           's2.resources.samplerState' ( temp sampler)
0:?           's1.samplerState' ( temp sampler)
0:30        move second child to first child ( temp int)
0:?           's2.resources.s0.x' ( temp int)
0:?           's1.s0.x' ( temp int)
0:30        move second child to first child ( temp int)
0:?           's2.resources.s0.y' ( temp int)
0:?           's1.s0.y' ( temp int)
0:30        move second child to first child ( temp sampler)
0:?           's2.resources.s0.ss' ( temp sampler)
0:?           's1.s0.ss' ( temp sampler)
0:30        move second child to first child ( temp int)
0:?           's2.resources.a' ( temp int)
0:?           's1.a' ( temp int)
0:31      Branch: Return with expression
0:?         Constant:
0:?           0.000000
0:?           0.000000
0:?           0.000000
0:?           0.000000
0:27  Function Definition: main( ( temp void)
0:27    Function Parameters: 
0:?     Sequence
0:27      move second child to first child ( temp 4-component vector of float)
0:?         'vpos' ( temp 4-component vector of float)
0:?         'vpos' (layout( location=0) in 4-component vector of float)
0:27      move second child to first child ( temp 4-component vector of float)
0:?         '@entryPointOutput' (layout( location=0) out 4-component vector of float)
0:27        Function Call: @main(vf4; ( temp 4-component vector of float)
0:?           'vpos' ( temp 4-component vector of float)
0:?   Linker Objects
0:?     '@entryPointOutput' (layout( location=0) out 4-component vector of float)
0:?     'vpos' (layout( location=0) in 4-component vector of float)


Linked fragment stage:


Shader version: 500
gl_FragCoord origin is upper left
0:? Sequence
0:27  Function Definition: @main(vf4; ( temp 4-component vector of float)
0:27    Function Parameters: 
0:27      'vpos' ( in 4-component vector of float)
0:?     Sequence
0:30      Sequence
0:30        move second child to first child ( temp float)
0:?           's2.resources.b' ( temp float)
0:?           's1.b' ( temp float)
0:30        move second child to first child ( temp sampler)
0:?           's2.resources.samplerState' ( temp sampler)
0:?           's1.samplerState' ( temp sampler)
0:30        move second child to first child ( temp int)
0:?           's2.resources.s0.x' ( temp int)
0:?           's1.s0.x' ( temp int)
0:30        move second child to first child ( temp int)
0:?           's2.resources.s0.y' ( temp int)
0:?           's1.s0.y' ( temp int)
0:30        move second child to first child ( temp sampler)
0:?           's2.resources.s0.ss' ( temp sampler)
0:?           's1.s0.ss' ( temp sampler)
0:30        move second child to first child ( temp int)
0:?           's2.resources.a' ( temp int)
0:?           's1.a' ( temp int)
0:31      Branch: Return with expression
0:?         Constant:
0:?           0.000000
0:?           0.000000
0:?           0.000000
0:?           0.000000
0:27  Function Definition: main( ( temp void)
0:27    Function Parameters: 
0:?     Sequence
0:27      move second child to first child ( temp 4-component vector of float)
0:?         'vpos' ( temp 4-component vector of float)
0:?         'vpos' (layout( location=0) in 4-component vector of float)
0:27      move second child to first child ( temp 4-component vector of float)
0:?         '@entryPointOutput' (layout( location=0) out 4-component vector of float)
0:27        Function Call: @main(vf4; ( temp 4-component vector of float)
0:?           'vpos' ( temp 4-component vector of float)
0:?   Linker Objects
0:?     '@entryPointOutput' (layout( location=0) out 4-component vector of float)
0:?     'vpos' (layout( location=0) in 4-component vector of float)

// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 49

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 42 45
                              ExecutionMode 4 OriginUpperLeft
                              Source HLSL 500
                              Name 4  "main"
                              Name 11  "@main(vf4;"
                              Name 10  "vpos"
                              Name 14  "s2.resources.b"
                              Name 15  "s1.b"
                              Name 19  "s2.resources.samplerState"
                              Name 20  "s1.samplerState"
                              Name 24  "s2.resources.s0.x"
                              Name 25  "s1.s0.x"
                              Name 27  "s2.resources.s0.y"
                              Name 28  "s1.s0.y"
                              Name 30  "s2.resources.s0.ss"
                              Name 31  "s1.s0.ss"
                              Name 33  "s2.resources.a"
                              Name 34  "s1.a"
                              Name 40  "vpos"
                              Name 42  "vpos"
                              Name 45  "@entryPointOutput"
                              Name 46  "param"
                              Decorate 42(vpos) Location 0
                              Decorate 45(@entryPointOutput) Location 0
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypeVector 6(float) 4
               8:             TypePointer Function 7(fvec4)
               9:             TypeFunction 7(fvec4) 8(ptr)
              13:             TypePointer Function 6(float)
              17:             TypeSampler
              18:             TypePointer Function 17
              22:             TypeInt 32 1
              23:             TypePointer Function 22(int)
              36:    6(float) Constant 0
              37:    7(fvec4) ConstantComposite 36 36 36 36
              41:             TypePointer Input 7(fvec4)
        42(vpos):     41(ptr) Variable Input
              44:             TypePointer Output 7(fvec4)
45(@entryPointOutput):     44(ptr) Variable Output
         4(main):           2 Function None 3
               5:             Label
        40(vpos):      8(ptr) Variable Function
       46(param):      8(ptr) Variable Function
              43:    7(fvec4) Load 42(vpos)
                              Store 40(vpos) 43
              47:    7(fvec4) Load 40(vpos)
                              Store 46(param) 47
              48:    7(fvec4) FunctionCall 11(@main(vf4;) 46(param)
                              Store 45(@entryPointOutput) 48
                              Return
                              FunctionEnd
  11(@main(vf4;):    7(fvec4) Function None 9
        10(vpos):      8(ptr) FunctionParameter
              12:             Label
14(s2.resources.b):     13(ptr) Variable Function
        15(s1.b):     13(ptr) Variable Function
19(s2.resources.samplerState):     18(ptr) Variable Function
20(s1.samplerState):     18(ptr) Variable Function
24(s2.resources.s0.x):     23(ptr) Variable Function
     25(s1.s0.x):     23(ptr) Variable Function
27(s2.resources.s0.y):     23(ptr) Variable Function
     28(s1.s0.y):     23(ptr) Variable Function
30(s2.resources.s0.ss):     18(ptr) Variable Function
    31(s1.s0.ss):     18(ptr) Variable Function
33(s2.resources.a):     23(ptr) Variable Function
        34(s1.a):     23(ptr) Variable Function
              16:    6(float) Load 15(s1.b)
                              Store 14(s2.resources.b) 16
              21:          17 Load 20(s1.samplerState)
                              Store 19(s2.resources.samplerState) 21
              26:     22(int) Load 25(s1.s0.x)
                              Store 24(s2.resources.s0.x) 26
              29:     22(int) Load 28(s1.s0.y)
                              Store 27(s2.resources.s0.y) 29
              32:          17 Load 31(s1.s0.ss)
                              Store 30(s2.resources.s0.ss) 32
              35:     22(int) Load 34(s1.a)
                              Store 33(s2.resources.a) 35
                              ReturnValue 37
                              FunctionEnd
